hack/generator/pkg/astmodel/map_type.go

```diff
@@ -62,12 +62,12 @@ func (m *MapType) Equals(t Type) bool {
 	return false
 }
 
-// CreateRelatedDefinitions returns any definitions required by our key or value types
-func (m *MapType) CreateRelatedDefinitions(ref PackageReference, namehint string, idFactory IdentifierFactory) []Definition {
-	result := m.key.CreateRelatedDefinitions(ref, namehint, idFactory)
-
-	otherDefinitions := m.value.CreateRelatedDefinitions(ref, namehint, idFactory)
-	result = append(result, otherDefinitions...)
+func (m *MapType) CreateInternalDefinitions(name *TypeName, idFactory IdentifierFactory) (Type, []TypeDefiner) {
+	newKeyType, otherTypes := m.key.CreateInternalDefinitions(name, idFactory)
+	newValueType, otherTypes2 := m.value.CreateInternalDefinitions(name, idFactory)
```

Comment:
Suggestions for variable naming ...
```
keyType, otherKeyTypes := ...
valueType, otherValueTypes := ...
```

Reply:
went with `keyOtherTypes` and `valueOtherTypes`